src/canister_tests/src/tests.rs

```diff
@@ -66,6 +66,165 @@ fn registration_with_mismatched_sender_fails() {
     );
 }
 
+/// Tests for the HTTP interactions according to the HTTP gateway spec: https://internetcomputer.org/docs/current/references/ic-interface-spec/#http-gateway
+#[cfg(test)]
+mod http_tests {
+    use crate::certificate_validation::validate_certification;
+    use crate::framework::CallError;
+    use crate::{api, framework};
+    use ic_state_machine_tests::StateMachine;
+    use internet_identity_interface::{HeaderField, HttpRequest};
+    use itertools::Itertools;
+    use regex::Regex;
+    use serde_bytes::ByteBuf;
+
+    #[test]
+    fn ii_canister_serves_http_assets() -> Result<(), CallError> {
+        let assets: Vec<(&str, Option<&str>)> = vec![
+            ("/", None),
+            ("/index.html", None),
+            ("/index.js", Some("gzip")),
+            ("/loader.webp", None),
+            ("/favicon.ico", None),
+            ("/ic-badge.svg", None),
+        ];
+        let env = StateMachine::new();
+        let canister_id = framework::install_ii_canister(&env, framework::II_WASM_PREVIOUS.clone());
```

Comment:
do you really want "II_WASM_PREVIOUS" here?

Reply:
No, thanks. Also, I have just realized, this is wrong in a lot of other places too --> PR incoming.